mb_netmgmt/ssh.py

```diff
@@ -88,19 +89,24 @@ def respond(self, response, request_id):
         return response
 
     def read_proxy_response(self):
-        message = self.read_message(
-            self.channel.upstream, [self.channel.command_prompt]
-        )
+        patterns = []
+        patterns.append(self.channel.command_prompt)  # Initial command prompt
+        patterns.append(b"[\r\n\x1b\[K][\-\w+\.:/]+(?:\([^\)]+\))?[>#] ?$")  # IOS
+        patterns.append(b"[\r\n\x00\x1b\[K]RP/\d+/(?:RS?P)?\d+\/CPU\d+:[^#]+(?:\([^\)]+\))?#$")  # IOS XR
+        patterns.append(b"[\r\n\x00\x1b\[K](?P<text>[\w/ .:,\(\)\-\?]*)(?P<default>\[[\w/.:\-]*\])?(?(default)(?P<end1>(?:\?|: ?|)$)|(?P<end2>: $))")  # Interactive prompt
+        patterns.append(b"[\r\n\x00\x1b\[K] --More-- $")  # Terminal paging
```

Comment:
I assume that you didn't hand-craft these patterns by yourself ;)
Do you maybe have a source where we can refer to in a comment or maybe even use this source directly?

Also it would be nice to not have all the patterns duplicated, but instead to have a central place.

Reply:
The IOS pattern comes from [here](https://github.com/knipknap/exscript/blob/master/Exscript/protocols/drivers/ios.py#L35), and the XR pattern from [here](https://github.com/knipknap/exscript/blob/master/Exscript/protocols/drivers/ios_xr.py#L32) (with an additional escape code `x1b[K`).
The other two are new ones :)
What's your preferred way for a central place for the patterns?

Reply:
@christian-sahlmann I've made some updates based on your input, please check 🙂